packages/backend/src/apps/lettersg/actions/create-letter/index.ts

```diff
@@ -115,9 +125,21 @@ const action: IRawAction = {
         return
       }
 
+      const { data: templateData } = await $.http.get(
+        '/v1/templates/:templateId',
+        {
+          urlPathParams: {
+            templateId: $.step.parameters.templateId,
+          },
+        },
+      )
```

Comment:
this API is called 3 times in different files, could you refactor this with proper handling into a common function

Reply:
ok got it